vendor/bat-native-ledger/include/bat/ledger/public/interfaces/ledger.mojom

```diff
@@ -111,13 +111,23 @@ struct VisitData {
   string favicon_url;
 };
 
+struct StringSet {
+  array<string> elements;
+};
+
+struct Regions {
+  StringSet allow;
```

Comment:
I see that `StringSet` (`std::set` when mapped) is the more appropriate data structure, but it seems the codebase would be simpler if this were just `array<string>`. If we're not really getting much benefit from the `std::set` mapping, maybe we should just use `array<string>`?

Reply:
I can see benefit in using `std::set<std::string>`. Using `array<string>` wouldn't make the code simpler, as `base::Value::List` (the type we construct from `std::set<std::string>`) doesn't have a constructor, which could directly utilize the default mapping (`std::vector<std::string>`). It's only the actual mojom type mapping that's not straightforward to set up, but that's already done, so we can even add other type mappings in the future easily. Also, the cool part is: you can specify that it's `move_only`. I'd stick with it, if you don't mind. 🙂

Reply:
> Using array<string> wouldn't make the code simpler

We wouldn't have to maintain custom type mappings, right? That's a good thing. Are there any other advantages besides move-only?

Reply:
Totally fine if you want to keep it though - just trying to understand the options and follow the "keep it simple" approach. 👍

Reply:
So, I mean, Mojo lacks a `set<>` type (whereas they have a proper default type mapping for `map<>` for some strange reason), which is annoying in situations where in C++ you wouldn't use a `std::vector<>` to store the data — and this wouldn't be the first place in code where we would choose the wrong data structure because of Mojo's defaults. That said, in this particular case we don't need any of the lists sorted, and we don't care about the uniqueness of the country codes (as it doesn't hurt if a country code shows up more than once), so we can get by with just using `array<string>`, but in the long-term, I would consider using our own type mappings as we see fit. There wouldn't be much to maintain in the `*mojom_traits.h/.cc` files, in my opinion.